src/pgReal.ts

```diff
@@ -98,7 +100,10 @@ export async function withClient<T = void>(
   parsedSettings: ParsedSettings,
   callback: (pgClient: PoolClient, context: Context) => Promise<T>,
 ): Promise<T> {
-  const details = getPoolDetailsFromConnectionString(connectionString);
+  const details = getPoolDetailsFromConnectionString(
+    connectionString,
+    parsedSettings,
+  );
```

Comment:
I've just remembered we have a convention of always passing `parsedSettings` as the first argument.

```suggestion
  const details = getPoolDetailsFromConnectionString(
    parsedSettings,
    connectionString,
  );
```